src/modules/content-seeding/news-aggregator.js

```diff
@@ -0,0 +1,290 @@
+	return articlePubDate > lastupdatetime;
+}
+
+function insertIntoFeeds (url: string): Promise<Array<{ rowCount: number }>> {
+	return performWriteQuery([ {
+		$: `INSERT INTO feeds (url) 
+			VALUES ( &{url} )`,
+		url
+	} ]);
+}
+
+function insertNewArticles (url: string, article: Article, contentForLexemes: string): Promise<Array<{ rowCount: number }>> {
+	return performWriteQuery([ {
+		$: `INSERT INTO articles (
+			url, rawjson, terms
+		) VALUES (
+			&{url}, &{rawjson}, to_tsvector(&{contentForLexemes}) 
+		)`,
+		url,
+		rawjson: JSON.stringify(article),
+		contentForLexemes
+	} ]);
+}
+
+function parseNewsFeed (url: string): Promise<Article> {
+	return parseFeed(url);
+}
+
+function truncateArticlesTable (): Promise<any> {
+	return performReadQuery({
+		$: 'TRUNCATE TABLE articles'
+	});
+}
+
+function updateFeedsOnNewArticles (lastupdatetime: number, url: string, mtbu: number): Promise<Array<{ rowCount: number }>> {
+	return performWriteQuery([ {
+		$: `UPDATE feeds SET lastrequesttime = extract(epoch from NOW()) * 1000,
+			lastupdatetime = &{lastupdatetime}, mtbu = &{mtbu} 
+			WHERE url=&{url}`,
+		lastupdatetime,
+		url,
+		mtbu
+	} ]);
+}
+
+function updateFeedsOnNoNewArtcles (url: string): Promise<Array<{ rowCount: number }>> {
+	return performWriteQuery([ {
+		$: 'UPDATE feeds SET lastrequesttime = extract(epoch from NOW()) * 1000 WHERE url=&{url}',
+		url
+	} ]);
+}
+
+export const newsAggregator = (): Promise<Array<any>> => {
+	// Check if the feeds table is not empty.
+	return performReadQuery({
+		$: 'SELECT url FROM feeds',
+	})
+	.then((results) => {
+		// results: [{url: _}, ...]
+		if (results.length === 0) {
+			winston.info('feeds table is empty');
+			// insert seed url if feeds table is empty
+			const seedUrl = 'http://bangalore.citizenmatters.in/rss';
+			return insertIntoFeeds(seedUrl);
+		}
+		return [ { rowCount: 0 } ];
+	})
+	.then(async (response) => {
+		// response: [{.., rowCount: _, ...}, ..]
+		if (response[0].rowCount === 1) {
+			winston.info('Insert operation performed successfully.');
+		}
+		await truncateArticlesTable();
+		// feeds: [{url: _, mtbu: _, lastrequesttime: _, lastupdatetime: _}, .....]
+		const feeds = await filterFeeds();
+		if (feeds.length > 0) {
+			const parseFeedPromises = [];
+			for (const feed of feeds) {
+				/*
+					CHECK FOR NEW ARTICLES:
+						- checks:
+							-> shouldn't be older than 1 week.
+							-> Avoid duplicate stroies from the same feed.
+							-> To avoid duplicate stories, compare article's pubdate with the lastupdatetime for that feed.
+				*/
+				const newArticles = [];
+				let latestArticleDate = feed.lastupdatetime;
+				parseFeedPromises.push(
+					parseNewsFeed(feed.url)
+					.then((newsArticles) => {
+						const promises = [];
+						for (const newsArticle of newsArticles) {
+							const articlePubDate = (new Date(newsArticle.published)).getTime();
+							const sevenDays = 7 * 24 * 60 * 60 * 1000;
```

Comment:
Put such constants in all caps at the top. Also rename it to what it does rather than what its value is: when we change it to 14 days we don't want a situation where `const sevenDays = <14 days>`.

E.g. `const MAX_ARTICLE_AGE = 7 * 24 * 60 * 60 * 1000; // No article older than this will ever be posted.`